packages/react/src/table/ScrollerContext.tsx

```diff
@@ -0,0 +1,29 @@
+import { createContext, Dispatch, SetStateAction, useContext } from 'react';
+
+type OverlayOffsets = {
+	bottom?: 0 | `${number}rem`;
+	left?: 0 | `${number}rem`;
+	right?: 0 | `${number}rem`;
+	top?: 0 | `${number}rem`;
+};
+
+export type ScrollerContextType = {
+	setOverlayOffsets?: Dispatch<SetStateAction<OverlayOffsets | undefined>>;
+	overlayOffsets?: OverlayOffsets;
+};
+
+export const ScrollerContext = createContext<ScrollerContextType | undefined>(
+	undefined
+);
+
+export function useScrollerContext() {
+	const context = useContext(ScrollerContext);
+
+	if (typeof context === 'undefined') {
+		throw Error(
+			'ScrollerContext not found. If using a Table, ensure it is wrapped with <TableWrapper>.'
+		);
```

Comment:
I'm a little concerned by this because I'm not confident that all consumers are currently using the `<TableWrapper>` 🤔

Reply:
Yep, have just checked and it's not always used…